tests/load/gcp/prom-query-locustfile.tpl.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python3
+
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import json
+import random
+from datetime import datetime
+from time import time_ns
+
+from locust import HttpUser, TaskSet, constant_pacing, task
+from locust.contrib.fasthttp import FastHttpUser
+
+
+def timepad(pad: float):
+    """Decorator that guarantees a function won't return before some minimum time has elapsed.
+    Args:
+        pad: minimum time in seconds that should elapse before the wrapped function returns.
+    """
+    pad = max(0, pad)
+    def decorator(func):
+        import time
+        from functools import wraps
+        @wraps(func)
+        def inner(*args, **kwargs):
+            tic = time.time()
+            func(*args, **kwargs)
+            toc = time.time()
+            time.sleep(max(0, pad - (toc - tic)))
+        return inner
+    return decorator
+
+
+def scale_probabilities(*args):
+    """Scale a sequence of occurrences into probabilities that sum up to 1."""
+    scale_factor = 1.0 / sum(args)
+    return [scale_factor * arg for arg in args]
+
+def probabilities_to_weights(*args):
+    """Scale a sequence of probabilities into integer weights."""
+    scale_factor = 10.0 / min(args)
+    return [int(scale_factor * arg) for arg in args]
+
+
+GRAFANA_DASHBOARD_REFRESH_PERIOD = ${REFRESH_INTERVAL}  # seconds
+
+# simulate a "full timeseries" fetch (large query) every 10 minutes (per panel)
+entire_panel_fetch_period = 10 * 60  # 10 minutes
+incremental_fetch_period = GRAFANA_DASHBOARD_REFRESH_PERIOD
+
+entire_panel_fetch_weight, incremental_fetch_weight = probabilities_to_weights(
```

Comment:
Judging by indentation these statements seem to be in module scope. If so positioning them between function definitions makes this module harder to read. Typically such statements are at beginning or ending of module or even better guarded by a `if __name__ == "__main__"` or in a `main()` function.